hazelcast/src/main/java/com/hazelcast/sql/impl/SqlServiceImpl.java

```diff
@@ -209,6 +214,12 @@ public SqlResult execute(
                     statement.getExpectedResultType(),
                     securityContext
             );
+            if (!skipStats && sqlResult instanceof AbstractSqlResult) {
```

Comment:
You may introduce `isInfiniteRows()` in `SqlResult` interface, since it is implemented in each subclass.

Reply:
changed it!